src/occwl/edge.py

```diff
@@ -452,3 +457,46 @@ def find_left_and_right_faces(self, faces):
             right_face = face1
 
         return left_face, right_face
+
+    def get_polyline(self, deflection=0.0005, algorithm="QuasiUniformDeflection"):
+        """
+        Get a polyline, represented as a sequence of points, from this edge
+
+        Args:
+            deflection (float): Lower deflection results in more precision
+                                and therefore more points
+            algorithm (string): Algorithm to use, can be one of:
+                                QuasiUniformDeflection, UniformAbscissa,
+                                or UniformDeflection
+        
+        Returns:
+            2D np.ndarray: Points
+        """
+        # If we don't have a valid curve, return an empty array
+        if not self.has_curve():
+            return np.empty(shape=(0,0), dtype=np.float32)
+        
+        curve_adaptor = BRepAdaptor_Curve(self.topods_shape())
+        first_param = curve_adaptor.FirstParameter()
+        last_param = curve_adaptor.LastParameter()
+
+        if algorithm == "QuasiUniformDeflection":
+            discretizer = GCPnts_QuasiUniformDeflection()
+        elif algorithm == "UniformAbscissa":
+            discretizer = GCPnts_UniformAbscissa()
+        elif algorithm == "UniformDeflection":
+            discretizer = GCPnts_UniformDeflection()
+        else:
+            raise Exception("Unknown algorithm")
+        discretizer.Initialize(curve_adaptor, deflection, first_param, last_param)
+
+        if not discretizer.IsDone():
+            raise Exception("Discretizer not done.")
+        if not discretizer.NbPoints() > 0:
+            raise Exception("Discretizer nb points not > 0.")
+
+        points = []
+        for i in range(1, discretizer.NbPoints() + 1):
+            p = curve_adaptor.Value(discretizer.Parameter(i))
+            points.append(np.array(list(p.Coord())))
+        return np.asarray(points, dtype=np.float32)
```

Comment:
Can we directly just call `discretize_edge()` inside this method? Or did you have to make any changes to that code?

Reply:
So we have different error handling (returning an empty `np.array` when there is not a curve) and also pack everything into an `np.array` and return that.